src/components/Icon/Icon.scss

```diff
@@ -27,192 +27,394 @@
 }
 
 .isColored {
-  color: color('white');
+  color: var(--p-text-on-interactive, color('white'));
 }
 
 .colorWhite {
   @include color-icon('white');
   color: transparent;
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorBlack {
   @include color-icon('black');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorSkyLighter {
   @include color-icon('sky', 'lighter');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorSkyLight {
   @include color-icon('sky', 'light');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorSky {
   @include color-icon('sky');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorSkyDark {
   @include color-icon('sky', 'dark');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorInkLightest {
   @include color-icon('ink', 'lightest');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorInkLighter {
   @include color-icon('ink', 'lighter');
 
   &::before {
-    background-color: color('sky');
+    background-color: var(--p-surface, color('sky'));
+  }
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
   }
 }
 
 .colorInkLight {
   @include color-icon('ink', 'light');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorInk {
   @include color-icon('ink');
 
   &::before {
-    background-color: color('sky');
+    background-color: var(--p-surface, color('sky'));
+  }
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
   }
 }
 
 .colorBlueLighter {
   @include color-icon('blue', 'lighter');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorBlueLight {
   @include color-icon('blue', 'light');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorBlue {
   @include color-icon('blue');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
 
 .colorBlueDark {
   @include color-icon('blue', 'dark');
 
-  &::before {
-    background-color: color('blue', 'light');
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
   }
 }
 
 .colorBlueDarker {
   @include color-icon('blue', 'darker');
+
+  &.newDesignLanguage {
+    @include recolor-icon(var(--p-icon));
+  }
 }
```

Comment:
In conversations with @sarahill, she mentioned wanting to discourage the use of icon-only buttons, so I have omitted an `interactive` icon color, and aliased all blue icons to `--p-icon` in the new design langauge.

Reply:
Blue icons in icon-only buttons are weird anyways imo.